src/Searchlight/LinqExecutor.cs

```diff
@@ -45,8 +45,35 @@ public static IEnumerable<T> QueryCollection<T>(this SyntaxTree tree, IEnumerabl
                     (from sort in tree.OrderBy select $"{sort.Column.FieldName} {sort.DirectionStr()}"));
                 queryable = queryable.OrderBy(sortExpression);
             }
+            
+            // If the user requested pagination
+            switch ((tree.PageNumber, tree.PageSize))
+            {
+                // case 1: user specified page number and page size
+                case (> 0, > 0):
+                    queryable = queryable.Skip((int) (tree.PageSize * tree.PageNumber)).Take((int) tree.PageSize);
+                    
+                    break;
+                
+                // case 2: user specified a page number but no page size
+                case (> 0, null):
+                    queryable = queryable.Skip((int) tree.PageNumber * 200).Take(200);
+
+                    break;
+                
+                // case 3: user specified a page size but no page number
+                case (null, > 0):
+                    queryable = queryable.Take((int) tree.PageSize);
+
+                    break;
+                
+                // catch all: take the first 200 entries
+                default:
```

Comment:
This seems wrong.  In the default case, we should not call Take() - we should just return everything.  We want people to be able to support fetch all if they want.

Reply:
Updated!